Fixes for Coverity defects in gradient serialization

peter-b opened the pull request.

montegoulding commented on engine/src/gradient.cpp:
```diff
@@ -665,15 +665,16 @@ IO_stat MCGradientFillSerialize(MCGradientFill *p_gradient, MCObjectOutputStream
-			t_stat = p_stream . WriteS16(t_point[i] . y);
-	}
+
+    auto t_write_point = [&p_stream, &t_stat](MCPoint& p) {
```
@peter-b this is neat (I had to google lambda expressions 😞 ) however I wonder if we should move it to `MCObjectOutputStream::WritePoint` for reuse?

peter-b replied:
@montegoulding Maybe we should find a good general solution for extending what can be written to an `MCObjectOutputStream` so that the code for serialising a particular type can live near the other functions that work on that type.  However, that is beyond the scope of what I was trying to achieve with this PR! 😉

runrevmark replied:
Pragmatically the number of things that might be written to an MCObjectStream is quite small - so adding a WriteMCPoint to MCObjectStrem seems perfectly sensible, to me at least :)

peter-b committed:
[CID 16371] Remove non-MCObjectOutputStream gradient fill serialization

This implementation is unused.

Coverity-ID: 16371

peter-b committed:
[CID 16268] Use MCObjectStream::WritePoint in MCGradientFillSerialize()

For convenience, `MCGradientFillSerialize()` previously used some
pointer magic and a loop to serialize points.  This confused static
analysis with Coverity, and was arguably a strict aliasing violation.

Coverity-ID: 16268

peter-b committed:
Add MCObjectStream::WritePoint()

Only used in one place, so add it to the header.

peter-b commented:
@montegoulding @runrevmark Done.

runrevmark commented:
@livecode-vulcan review ok 55c713c

livecode-vulcan commented:
💙 review by @runrevmark ok 55c713ca27223d2ed05c90ba0cf6b4df87b8b520

livecode-vulcan commented:
😎 test success 55c713ca27223d2ed05c90ba0cf6b4df87b8b520

* try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/914)
* try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/920)
* try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/912)
* try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/913)
* try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/920)
* try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/911)
* try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/911)
